clang/test/ClangScanDeps/include-tree-multiple-commands.c

```diff
@@ -0,0 +1,200 @@
+
+// RUN: %deps-to-rsp %t/deps.1.json --module-name=Mod             > %t/Mod.1.rsp
+// RUN: %deps-to-rsp %t/deps.1.json --tu-index 0 --tu-cmd-index 0 > %t/tu-cpp.1.rsp
+// RUN: %deps-to-rsp %t/deps.1.json --tu-index 0 --tu-cmd-index 1 > %t/tu-emit-ir.1.rsp
+// RUN: %deps-to-rsp %t/deps.1.json --tu-index 0 --tu-cmd-index 2 > %t/tu-emit-asm.1.rsp
+// RUN: %deps-to-rsp %t/deps.1.json --tu-index 0 --tu-cmd-index 3 > %t/tu-cc1as.1.rsp
+// RUN: %clang @%t/Mod.1.rsp         -Rcompile-job-cache 2>&1 | FileCheck %s -check-prefix=CACHE-HIT
+// RUN: %clang @%t/tu-cpp.1.rsp      -Rcompile-job-cache 2>&1 | FileCheck %s -check-prefix=CACHE-MISS
+// RUN: %clang @%t/tu-emit-ir.1.rsp  -Rcompile-job-cache 2>&1 | FileCheck %s -check-prefix=CACHE-HIT
+// RUN: %clang @%t/tu-emit-asm.1.rsp -Rcompile-job-cache 2>&1 | FileCheck %s -check-prefix=CACHE-HIT
+// RUN: %clang @%t/tu-cc1as.1.rsp
+// RUN: mv %t/tu.i  %t/tu.1.i
+// RUN: mv %t/tu.bc %t/tu.1.bc
+// RUN: mv %t/tu.s  %t/tu.1.s
+// RUN: mv %t/tu.o  %t/tu.1.o
+
+// RUN: diff %t/tu.1.i  %t/tu.0.i
+// RUN: diff %t/tu.1.bc %t/tu.0.bc
+// RUN: diff %t/tu.1.s  %t/tu.0.s
+// RUN: diff %t/tu.1.o  %t/tu.0.o
+
+// CACHE-HIT: remark: compile job cache hit
+// CACHE-MISS: remark: compile job cache miss
+
+// CHECK:      "modules": [
+// CHECK-NEXT:   {
+// CHECK-NEXT:     "cache-key": "[[M_CACHE_KEY:llvmcas://[[:xdigit:]]+]]"
+// CHECK-NEXT:     "cas-include-tree-id": "[[M_INCLUDE_TREE:llvmcas://[[:xdigit:]]+]]"
+// CHECK-NEXT:     "clang-module-deps": []
+// CHECK-NEXT:     "clang-modulemap-file": "[[PREFIX]]/module.modulemap"
+// CHECK-NEXT:     "command-line": [
+// CHECK:            "-fcas-include-tree"
+// CHECK-NEXT:       "[[M_INCLUDE_TREE]]"
+// CHECK:          ]
+// CHECK:          "name": "Mod"
+// CHECK-NEXT:   }
+// CHECK-NEXT: ]
+// CHECK-NEXT: "translation-units": [
+// CHECK-NEXT:   {
+// CHECK:          "commands": [
+// CHECK-NEXT:       {
+// CHECK-NEXT:         "cache-key": "[[CPP_CACHE_KEY:llvmcas://[[:xdigit:]]+]]"
+// CHECK-NEXT:         "cas-include-tree-id": "[[CPP_INCLUDE_TREE:llvmcas://[[:xdigit:]]+]]"
+// CHECK-NEXT:         "clang-context-hash": "{{.*}}"
+// CHECK-NEXT:         "clang-module-deps": [
+// CHECK-NEXT:           {
+// CHECK-NEXT:             "context-hash": "{{.*}}
+// CHECK-NEXT:             "module-name": "Mod"
+// CHECK-NEXT:           }
+// CHECK-NEXT:         ]
+// CHECK-NEXT:         "command-line": [
+// CHECK-NEXT:           "-cc1"
+// CHECK:                "-o"
+// CHECK-NEXT:           "[[PREFIX]]/tu.i"
+// CHECK:                "-fcas-include-tree"
+// CHECK-NEXT:           "[[CPP_INCLUDE_TREE]]"
+// CHECK-NOT:            "-finput-file-cache-key"
+// CHECK:                "-E"
+// CHECK:                "-fmodule-file-cache-key"
+// CHECK-NEXT:           "[[PREFIX]]/modules/{{.*}}/Mod-{{.*}}.pcm"
+// CHECK-NEXT:           "[[M_CACHE_KEY]]"
+// CHECK:                "-fmodule-file={{.*}}[[PREFIX]]/modules/{{.*}}/Mod-{{.*}}.pcm"
+// CHECK:                "[[PREFIX]]/tu.c"
```

Comment:
What's happening here; I thought include tree commands don't have the input file name directly in the command-line.

Reply:
You're right. This check currently matches the file path in `"file-deps"`; `"command-line"` doesn't contain it at all. I must've accidentally copied this from some other test. I'll tighten the checks a little bit more.